### PR TITLE
[KYUUBI #5968] Support set authentication user for Trino engine

### LINKED ISSUE #5968: [Improvement] Support set authentication user for Trino engine
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Sometimes, when we connect to Trino using only one username and password, we should support setting the authentication user

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by lsm1**
# :mag: Description
## Issue References 🔗

This pull request fixes #5968 

## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Review by cxzl25 — APPROVED** [review]

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala [hunk 1]
```diff
@@ -141,7 +144,7 @@ class TrinoSessionImpl(
       require(
         serverScheme.equalsIgnoreCase("https"),
         "Trino engine using username/password requires HTTPS to be enabled")
-      builder.addInterceptor(OkHttpUtil.basicAuth(username, password))
+      builder.addInterceptor(OkHttpUtil.basicAuth(trinoUser, password))
```
Comment (pan3793) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:147]: The ClientSession still uses username(session user), will this cause issues?

Comment (lsm1) [reply externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:147]: > The ClientSession still uses `username`(session user), will this cause issues?

It only used for http verification, will have no other impact.

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala [hunk 2]
```diff
@@ -57,6 +57,9 @@ class TrinoSessionImpl(
   private val username: String = sessionConf
```
Comment (pan3793) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:57]: let's rename it to `sessionUser`

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala (hunk 1 above)
Comment (pan3793) [reply externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:147]: as trinoUser is only used for auth, move the definition to here, and rename variable to `user`

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**